src/apps/loyalty/containers/forgot_password/index.tsx

```diff
@@ -0,0 +1,146 @@
+// import * as fetch from "isomorphic-fetch"
+import * as React from "react"
+import styled from "styled-components"
+
+import Button from "../../../../components/buttons/inverted"
+import Icon from "../../../../components/icon"
+import Input from "../../../../components/input"
+import Message from "../../../../components/message"
+import Text from "../../../../components/text"
+import TextLink from "../../../../components/text_link"
+
+import colors from "../../../../assets/colors"
+
+const Container = styled.div`
+  padding: 20px;
+  max-width: 350px;
+  margin: 0 auto;
+  align-text: center;
+  display: flex;
+  align-items: center;
+  flex-direction: column;
+`
+
+const StyledInput = styled(Input)`
+  margin: 5px 0;
+  width: 100%;
+`
+
+interface Props {
+  submitEmailUrl: string,
+  appToken: string
+}
+
+interface State {
+  email: string
+  error?: boolean
+  showMessage: boolean
+  isEmailValid: boolean
+}
+
+class ForgotPasswordForm extends React.Component<Props, any> {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      email: "",
+      showMessage: false,
+      isEmailValid: false,
+    }
+  }
+
+  handleEmailFieldChange(e) {
+    const value = e.target.value
+
+    this.setState({
+      email: value,
+      isEmailValid: validateEmail(value),
+      showMessage: false,
+    })
+  }
+
+  onClickSubmitButton(e) {
+    const options: RequestInit = {
+      method: "POST",
+      // credentials: "same-origin",
+      headers: {
+        "Content-Type": "application/json",
+        "Accept": "application/json",
+        "X-XAPP-TOKEN": this.props.appToken,
+      },
+      body: JSON.stringify({
+        email: this.state.email,
+      }),
+    }
+
+    return fetch(this.props.submitEmailUrl, options).then(res => {
+      if (res.status === 201) {
+        this.setState({
+          showMessage: true,
+          error: false,
+        })
+      } else if (res.status === 400) { // Email Not Found
+        this.setState({
+          showMessage: true,
+          error: true,
+        })
+      }
+    })
+  }
+
+  renderMessageBox() {
+    const message = this.state.error
+          ? <span>No account exists for <b>{this.state.email}</b></span>
+          : <span>Instructions on how to reset your password have been sent to <b>{this.state.email}</b></span>
+
+    // console.log("rendering message", message)
+    return this.state.showMessage ? (
+      <Message error={!!this.state.error}>
+        {message}
+      </Message>
+    ) : ""
```

Comment:
It’s probably better to return `null` instead, if you don’t want anything rendered, because in RN this would lead to an error (it complains that any string should go in a `Text` element).